src/pages/Admin/VenueList/VenueList.tsx

```diff
@@ -3,21 +3,21 @@ import { Link } from "react-router-dom";
 
 import { isVenueWithRooms } from "types/venues";
 
-import { orderedVenuesSelector } from "utils/selectors";
 import { canHaveSubvenues } from "utils/venue";
 
-import { useSelector } from "hooks/useSelector";
+import { useRelatedVenues } from "hooks/useRelatedVenues";
 
 import { VenueListProps } from "./VenueList.types";
 
 const VenueList: React.FC<VenueListProps> = ({
   selectedVenueId,
   roomIndex,
 }) => {
-  // @debt This selector relies on all venues in firebase being loaded into memory.. not very efficient
-  const venues = useSelector(orderedVenuesSelector);
+  const { relatedVenues, isRelatedVenuesLoading } = useRelatedVenues({
+    currentVenueId: selectedVenueId,
+  });
 
-  if (!venues) return <>Loading...</>;
+  if (!isRelatedVenuesLoading) return <>Loading...</>;
```

Comment:
[NB] Would be good to replace this with `Loading` or `LoadingPage` as appropriate.

Reply:
replaced with `<Loading />` in 4944f3417cc3a961b2743f9c4db7316d816602dc